snapcraft/internal/lxd.py

```diff
@@ -304,24 +304,97 @@ def _ensure_container(self):
             check_call([
                 'lxc', 'config', 'set', self._container_name,
                 'raw.idmap', 'both {} 0'.format(os.getuid())])
+            # Remove existing device (to ensure we update old containers)
+            devices = self._get_container_status()['devices']
+            if self._project_folder in devices:
+                check_call([
+                    'lxc', 'config', 'device', 'remove', self._container_name,
+                    self._project_folder])
             check_call([
                 'lxc', 'start', self._container_name])
 
     def _setup_project(self):
         self._ensure_mount(self._project_folder, self._source)
 
+    def _get_container_address(self):
+        network = self._get_container_status()['state']['network']['eth0']
+        for address in network['addresses']:
+            if address['family'] == 'inet':
+                return address['address']
+        raise RuntimeError('No IP found for {}'.format(self._container_name))
```

Comment:
I changed to SnapcraftEnvironmentError like the other existing ones and [proposed a separate PR for a new error class](#1505).